pkg/virt-launcher/virtwrap/api/converter.go

```diff
@@ -1162,14 +1160,10 @@ func getCPUTopology(vmi *v1.VirtualMachineInstance) *CPUTopology {
 	}
 
 	if vmiCPU == nil || (vmiCPU.Cores == 0 && vmiCPU.Sockets == 0 && vmiCPU.Threads == 0) {
-		//if cores, sockets, threads are not set, take value from domain resources request or limits and
+		//if cores, sockets, threads are not set, take value from domain resources request and
 		//set value into sockets, which have best performance (https://bugzilla.redhat.com/show_bug.cgi?id=1653453)
-		resources := vmi.Spec.Domain.Resources
-		if cpuRequests, ok := resources.Requests[k8sv1.ResourceCPU]; ok {
-			sockets = uint32(cpuRequests.Value())
-		} else if cpuLimit, ok := resources.Limits[k8sv1.ResourceCPU]; ok {
-			sockets = uint32(cpuLimit.Value())
-		}
+		cpuRequests := vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceCPU]
+		sockets = uint32(cpuRequests.Value())
```

Comment:
I guess the whole setting default here is shaky, but would it make sense to keep the limit check (and give it priority over the requests amount?)

Reply:
I think not - from k8s documentation it seems that the memory limit is just a threshold that admins can set to say "if the memory consumed by the container exceeds that value, kill it". so I think admins would, for instance, set a VM with memory-request=2G and memory-limit=8G to say "the VM generally needs 2G to operate, if the memory consumption gets to 8G it is far higher than the expected value so it should be killed".

so if I understand this correctly, setting the guest topology according to the limit value is a bit awkward

Reply:
Hm there can be the following situation if we don't do this default: If we only have one socket, then we can never use more than one CPU at a time. So a request of 1 CPU and a limit of 10 CPUs can only ever use 1 CPU at max. If I am not mistaken.

Reply:
Removed my previous comment - it was stupid, I had memory rather than CPU limit in mind...

I see your point but note that on the other hand, as scheduling is done according to the requested amount, there's no guarantee that in that scenario we'll have more than 1 CPU available for the VM on the node - so if we'll set 10 sockets here and only 1 CPU would actually be allocated to the VM on the node - then the guest should somehow know it needs to avoid using more than 1 socket, right?

Anyway, as we discussed regarding memory once - I can't see a real-world scenario where users define VMs without specifying their CPU topology so I'm ok with any heuristic here.

Reply:
Done

Reply:
@ahadas yep the whole heuristic here is a little bit wacky. Maybe we can find a better one later.